src/classes/Extend/APICache.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Cache WooCommerce API requests.
+ *
+ * @package Niteo\WooCart\Defaults
+ */
+
+namespace Niteo\WooCart\Defaults\Extend;
+
+trait APICache {
+
+	/**
+	 * Endpoint to cache.
+	 *
+	 * @var string
+	 */
+	private $endpoint_to_cache = 'wc/v3/reports';
+
+	/**
+	 * Unique identifier for the cache.
+	 *
+	 * @var string
+	 */
+	private $cache_key;
+
+	/**
+	 * Cache key prefix.
+	 *
+	 * @var string
+	 */
+	private $cache_key_prefix = 'wc_reports_';
+
+	/**
+	 * Initialize API cache.
+	 *
+	 * @return void
+	 */
+	public function api_init() : void {
+		// Build cache key using API url.
+		$this->create_cache_key();
```

Comment:
Maybe not even add filters if URI does not contain the **wc/v3/reports**?

Reply:
Yup, the filters are not added if the `cache_key` is not present and the key is only set if all conditions for caching are satisfied.